### ISSUE TITLE
[BUG]: The path of assets is not correct

### ISSUE THREAD

**Issue by jinaiyuanbaojie** [FlutterGen/flutter_gen#322]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Command type

Dart command

### What happened?

The plugin of svg, lottie, flare and rive generate the wrong path of the related assets.
I have raised a PR https://github.com/FlutterGen/flutter_gen/pull/321.
Not sure if it is a bug. If I am wrong, I think we can add a feature to show the correct path of assets under packages. :)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wasabeef** [comment]
@jinaiyuanbaojie Thank you!

**Comment by wasabeef** [comment]
@jinaiyuanbaojie 
I just released `v5.0.3`.

**Comment by noinskit** [comment]
This just broke svg images in my app... `path` that was previously correctly `packages/package_name/images/misc/foo.svg` after upgrade became `packages/package_name/packages/package_name/images/misc/foo.svg` which obviously doesn't work. What am I doing wrong? In any case, 5.0.3 was definitely not a backward-compatible update.

**Comment by wasabeef** [comment]
@noinskit 
I implemented it with the intention of returning `packages/package_name/images/misc/foo.svg`, but is there a bug? 🤔

**Comment by noinskit** [comment]
From my perspective - yes, a bug. It worked for me before the change, stopped working after the change.

**Comment by jinaiyuanbaojie** [comment]
> This just broke svg images in my app... `path` that was previously correctly `packages/package_name/images/misc/foo.svg` after upgrade became `packages/package_name/packages/package_name/images/misc/foo.svg` which obviously doesn't work. What am I doing wrong? In any case, 5.0.3 was definitely not a backward-compatible update.

Hi man, So you add the prefix **packages/package_name** by yourself before. Am I right?
Could you post your codes here?

**Comment by noinskit** [comment]
I can't share the source directly, but what I'm doing is:
```
SvgPicture.asset(
  Assets.images.misc.foo.path,
  color: color,
  width: 30,
  package: packageName,
)
```
This is in the dependent package, in a class to be used by the top-level package.
I'm not manually concatenating any string prefixes. The concatenation based on `package` and `path` is done in the `SvgPicture` class.
If I use the `SvgGenImage.svg()` it's working.

In 5.0.3 it's actually not possible to get the `_assetName` without the package prefix anymore.

Note that `AssetGenImage` is totally different than `SvgGenImage`: it has a public `path` property (*without* the package prefix) and a `keyName` property (*with* the prefix).

**Comment by jinaiyuanbaojie** [comment]
> I can't share the source directly, but what I'm doing is:
> ```
> SvgPicture.asset(
>   Assets.images.misc.foo.path,
>   package: packageName,
> )
> ```
> This is in the dependent package, in a class to be used by the top-level package.
> I'm not manually concatenating any string prefixes. The concatenation based on `package` and `path` is done in the `SvgPicture` class.
> If I use the `SvgGenImage.svg()` it's working.
> 
> In 5.0.3 it's actually not possible to get the `_assetName` without the package prefix anymore.
> 
> Note that `AssetGenImage` is totally different than `SvgGenImage`: it has a public `path` property (*without* the package prefix) and a `keyName` property (*with* the prefix).

Got it.
- Assets.images.misc.foo.svg()
-
> ```
> SvgPicture.asset(
>   Assets.images.misc.foo.path
> )
> ```

Could you try one of them?
I prefer the first solution.

**Comment by noinskit** [comment]
@jinaiyuanbaojie I'm not sure if I understand. There are workarounds, like downgrading to 5.0.2 or avoiding `SvgGenImage.path`. It doesn't really change the nature of the bug.

**Comment by jinaiyuanbaojie** [comment]
> @jinaiyuanbaojie I'm not sure if I understand. There are workarounds, like downgrading to 5.0.2 or avoiding `SvgGenImage.path`. It doesn't really change the nature of the bug.

Yeah man. What matters is if it is a bug.
From my view, the path of assets should contains the package prefix, the absolute one.

**Comment by wasabeef** [comment]
We think this change is inherently correct, since the package prefix is included only if package_parameter_enabled is enabled.

**Comment by wasabeef** [comment]
@noinskit @jinaiyuanbaojie 
We think this change is inherently correct, since the package prefix is included only if package_parameter_enabled is enabled. However, I needed to be more specific about this intent in my CHANGELOG.

**Comment by wasabeef** [comment]
@noinskit @jinaiyuanbaojie 
What do you think about doing the following in `v5.1.0 (next version)` just like `asset.gen.dart`?

**When package_parameter_enabled = `true`**
```dart
String get path => _assetName;

String get keyName => 'packages/$packageName/\$_assetName';
```

**When package_parameter_enabled  = `false`**
```dart
String get path => _assetName;

String get keyName => _assetName;
```

**Comment by noinskit** [comment]
@wasabeef sure, this would work for me.

**Comment by jinaiyuanbaojie** [comment]
> @noinskit @jinaiyuanbaojie 
> What do you think about doing the following in `v5.1.0 (next version)` just like `asset.gen.dart`?
> 
> **When package_parameter_enabled = `true`**
> ```dart
> String get path => _assetName;
> 
> String get keyName => 'packages/$packageName/\$_assetName';
> ```
> 
> **When package_parameter_enabled  = `false`**
> ```dart
> String get path => _assetName;
> 
> String get keyName => _assetName;
> ```

@wasabeef @noinskit Yeah. That's fine:)

**Comment by wasabeef** [comment]
@jinaiyuanbaojie @noinskit 
Thank you for always supporting me. I just released the `v5.1.0`.

**Comment by jinaiyuanbaojie** [comment]
> @jinaiyuanbaojie @noinskit 
> Thank you for always supporting me. I just released the `v5.1.0`.

welcome. let's rock
